storage/Devices/Luks.h

```diff
@@ -52,8 +52,38 @@ namespace storage
 	 */
 	static std::vector<const Luks*> get_all(const Devicegraph* devicegraph);
 
+	/**
+	 * Get the LUKS UUID.
+	 */
 	const std::string& get_uuid() const;
 
+	/**
+	 * Get the LUKS label.
+	 *
+	 * Only available in LUKS version 2.
+	 */
+	const std::string& get_label() const;
+
+	/**
+	 * Set the LUKS label.
+	 *
+	 * Only available in LUKS version 2.
+	 */
+	void set_label(const std::string& label);
+
+	/**
+	 * Get extra options for luks format call.
+	 */
+	const std::string& get_format_options() const;
+
+	/**
+	 * Set extra options for luks format call.
+	 *
+	 * Options that modify the size of the resulting blk device
+	 * (e.g. --integrity) are not allowed.
+	 */
+	void set_format_options(const std::string& format_options);
```

Comment:
I do not know the general libstorage expectations but as the format options are literately injected into the command (no escaping/quoting) I think it should be mentioned in the comment that it's the responsibility of the caller to properly escape/quote the options.